packages/core/core-flows/src/cart/workflows/list-shipping-options-for-cart.ts

```diff
@@ -126,56 +172,49 @@ export const listShippingOptionsForCartWorkflow = createWorkflow(
 
         "calculated_price.*",
       ],
-      variables: {
-        context: {
-          is_return: !!input.is_return,
-          enabled_in_store: "true",
-        },
-        filters: {
-          fulfillment_set_id: fulfillmentSetIds,
-          address: {
-            city: cart.shipping_address?.city,
-            country_code: cart.shipping_address?.country_code,
-            province_code: cart.shipping_address?.province,
-          },
-        },
-
-        calculated_price: {
-          context: pricingContext,
-        },
-      },
+      variables: queryVariables,
     }).config({ name: "shipping-options-query" })
 
-    const shippingOptionsWithPrice = transform({ shippingOptions }, (data) => {
-      const optionsMissingPrices: string[] = []
+    const updatedShippingOptions = transform(
+      { shippingOptions, input },
+      ({ shippingOptions, input }) => {
+        const optionsMissingPrices: string[] = []
 
-      const options = data.shippingOptions.map((shippingOption) => {
-        const { calculated_price, ...options } = shippingOption ?? {}
+        const options = shippingOptions.map((shippingOption) => {
+          const { calculated_price, ...options } = shippingOption
 
-        if (options?.id && !isPresent(calculated_price?.calculated_amount)) {
-          optionsMissingPrices.push(options.id)
-        }
+          if (
+            shippingOption?.id &&
+            !isDefined(calculated_price?.calculated_amount)
+          ) {
+            optionsMissingPrices.push(options.id)
+          }
 
-        return {
-          ...options,
-          amount: calculated_price?.calculated_amount,
-          is_tax_inclusive:
-            !!calculated_price?.is_calculated_price_tax_inclusive,
-        }
-      })
+          return {
+            ...shippingOption,
+            amount: calculated_price?.calculated_amount,
+            is_tax_inclusive:
+              !!calculated_price?.is_calculated_price_tax_inclusive,
+          }
+        })
+
+        const shouldValidatePrices = input.options?.shouldValidatePrices ?? true
 
-      if (optionsMissingPrices.length) {
-        const ids = optionsMissingPrices.join(", ")
+        if (optionsMissingPrices.length && shouldValidatePrices) {
+          const ids = optionsMissingPrices.join(", ")
+
+          throw new MedusaError(
+            MedusaError.Types.INVALID_DATA,
+            `Shipping options with IDs ${ids} do not have a price`
+          )
+        }
 
-        throw new MedusaError(
-          MedusaError.Types.INVALID_DATA,
-          `Shipping options with IDs ${ids} do not have a price`
+        return options.filter((option) =>
+          isDefined(option?.calculated_price?.calculated_amount)
```

Comment:
thought: I believe this filter is redundant, as we would otherwise have thrown on undefined prices ealier

Reply:
depends on the validation comment here - https://github.com/medusajs/medusa/pull/10382/files#r1867735179